### PR TITLE
feat(node): implement TAPI signaling

### LINKED ISSUE #1921: Repurpose `BlockHeader.version` field for TAPI support signals

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Close #1921

### CODE REVIEW

**Comment by aesedepece** [comment]
I have made this PR target branch `tapi` instead of master.

File: node/src/actors/chain_manager/mod.rs
Comment (lrubiorod) [inline node/src/actors/chain_manager/mod.rs:2026]: This function should use the information of the TapiEngine of the other PR

File: node/src/actors/chain_manager/mod.rs
Comment (aesedepece) [inline node/src/actors/chain_manager/mod.rs:2021]: I can't see a need for activation ranges. Am I missing something?

File: node/src/actors/chain_manager/mod.rs
Comment (aesedepece) [inline node/src/actors/chain_manager/mod.rs:2016]: Let's run away from selling this as a voting / governance mechanism.

```suggestion
    fn tapi_signals_mask(&self, epoch: Epoch) -> u32 {
```

**Review by aesedepece — CHANGES_REQUESTED** [review]

File: node/src/actors/chain_manager/mod.rs
Comment (tmpolaczyk) [reply node/src/actors/chain_manager/mod.rs:2021]: This allows us to reuse bits for a different WIP later on. For example, bit 0 will be used for the activation of WIP0014 between epochs 500_000 and 700_000, but it will also be used for the activation of WIP0066 between epochs 12_000_000 and 13_000_000.

File: data_structures/src/mainnet_validations.rs
Comment (lrubiorod) [inline data_structures/src/mainnet_validations.rs:106]: for i in 0..tapi.len(){
    if let Some(bit_info) = tapi.get_mut(i, epoch) {
        if bit_info.wip == wip {
              return true;
        }
   }
}

false

File: config/src/config.rs
Comment (aesedepece) [inline config/src/config.rs:378]:
```suggestion
/// Allow miners to oppose activation of future protocol improvements even if their nodes
/// do implement the required logic. 
```

File: node/src/actors/messages.rs
Comment (aesedepece) [inline node/src/actors/messages.rs:706]:
```suggestion
    /// Active Witnet protocol improvements as of the current epoch.
```

File: node/src/actors/messages.rs
Comment (aesedepece) [inline node/src/actors/messages.rs:722]:
```suggestion
    /// Active Witnet protocol improvements as of the block that will include this tally.
```

**Review by aesedepece — CHANGES_REQUESTED** [review]
General comment: I'm OK with saying _wips_ in the code, but not in the documentation, where we should use naming that matches WIP-0014: _protocol improvements_

**Comment by tmpolaczyk** [comment]
> General comment: I'm OK with saying wips in the code, but not in the documentation, where we should use naming that matches WIP-0014: protocol improvements

Then we should review all the documentation before merging the tapi branch into master.

**Review by aesedepece — APPROVED** [review]

**Pull request merged**
